new thrift is triggering NPEs in tests. That is *bad*
```
Error:  org.apache.parquet.hadoop.thrift.TestParquetToThriftReadWriteAndProjection.testPullInRequiredLists -- Time elapsed: 0.027 s <<< ERROR!
org.apache.parquet.io.ParquetDecodingException: Can not read value at 1 in block 0 in file file:/home/runner/work/parquet-java/parquet-java/parquet-thrift/target/test/TestParquetToThriftReadWriteAndProjection/file.parquet
	at org.apache.parquet.hadoop.InternalParquetRecordReader.nextKeyValue(InternalParquetRecordReader.java:280)
	at org.apache.parquet.hadoop.ParquetRecordReader.nextKeyValue(ParquetRecordReader.java:207)
	at org.apache.parquet.hadoop.thrift.TestParquetToThriftReadWriteAndProjection.shouldDoProjection(TestParquetToThriftReadWriteAndProjection.java:375)
	at org.apache.parquet.hadoop.thrift.TestParquetToThriftReadWriteAndProjection.shouldDoProjectionWithThriftColumnFilter(TestParquetToThriftReadWriteAndProjection.java:337)
	at org.apache.parquet.hadoop.thrift.TestParquetToThriftReadWriteAndProjection.testPullInRequiredLists(TestParquetToThriftReadWriteAndProjection.java:301)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
Caused by: java.lang.NullPointerException
	at org.apache.thrift.protocol.TProtocol.incrementRecursionDepth(TProtocol.java:59)
	at org.apache.parquet.thrift.test.RequiredListFixture$RequiredListFixtureStandardScheme.read(RequiredListFixture.java:414)
	at org.apache.parquet.thrift.test.RequiredListFixture$RequiredListFixtureStandardScheme.read(RequiredListFixture.java:410)
	at org.apache.parquet.thrift.test.RequiredListFixture.read(RequiredListFixture.java:345)
	at org.apache.parquet.thrift.TBaseRecordConverter$1.readOneRecord(TBaseRecordConverter.java:63)
	at org.apache.parquet.thrift.TBaseRecordConverter$1.readOneRecord(TBaseRecordConverter.java:58)
	at org.apache.parquet.thrift.ThriftRecordConverter.getCurrentRecord(ThriftRecordConverter.java:945)
	at org.apache.parquet.io.RecordReaderImplementation.read(RecordReaderImplementation.java:439)
```

--- a/parquet-thrift/src/main/java/org/apache/parquet/thrift/BufferedProtocolReadToWrite.java
+++ b/parquet-thrift/src/main/java/org/apache/parquet/thrift/BufferedProtocolReadToWrite.java
@@ -579,7 +579,7 @@ public class BufferedProtocolReadToWrite implements ProtocolPipe {
   class NullProtocol extends TProtocol {
 
     public NullProtocol() {
-      super(null);
+      super(StubTTransport.INSTANCE);
     }
 
     @Override

--- a/parquet-thrift/src/main/java/org/apache/parquet/thrift/ParquetProtocol.java
+++ b/parquet-thrift/src/main/java/org/apache/parquet/thrift/ParquetProtocol.java
@@ -38,7 +38,7 @@ public abstract class ParquetProtocol extends TProtocol {
   private String name;
 
   ParquetProtocol() {
-    super(null);
+    super(StubTTransport.INSTANCE);
   }
 
   private String getClassInfo() {
@@ -54,7 +54,7 @@ public abstract class ParquetProtocol extends TProtocol {
    * @param name a meaningful debugging name for anonymous inner classes
    */
   public ParquetProtocol(String name) {
-    super(null);
+    super(StubTTransport.INSTANCE);
     this.name = name;
   }
 

--- a/parquet-thrift/src/main/java/org/apache/parquet/thrift/StubTTransport.java
+++ b/parquet-thrift/src/main/java/org/apache/parquet/thrift/StubTTransport.java
@@ -26,14 +26,14 @@ import org.apache.thrift.transport.TTransportException;
  * A stub transport which implements the minimum amount needed for range/depth
  * validation within the thrift library to succeed.
  */
-public final class StubTTransport extends TTransport {
+final class StubTTransport extends TTransport {
 
-  public static final StubTTransport INSTANCE = new StubTTransport();
+  static final StubTTransport INSTANCE = new StubTTransport();
 
   /**
    * There's no limits on recursion depth.
    */
-  private final TConfiguration conf = new TConfiguration(
+  private static final TConfiguration CONFIGURATION = new TConfiguration(
       TConfiguration.DEFAULT_MAX_FRAME_SIZE, TConfiguration.DEFAULT_MAX_MESSAGE_SIZE, Integer.MAX_VALUE);
 
   private StubTTransport() {}
@@ -66,7 +66,7 @@ public final class StubTTransport extends TTransport {
 
   @Override
   public TConfiguration getConfiguration() {
-    return conf;
+    return CONFIGURATION;
   }
 
   @Override

--- a/parquet-thrift/src/main/java/org/apache/parquet/thrift/StubTTransport.java
+++ b/parquet-thrift/src/main/java/org/apache/parquet/thrift/StubTTransport.java
@@ -34,16 +34,12 @@ public final class StubTTransport extends TTransport {
    * There's no limits on recursion depth.
    */
   private final TConfiguration conf = new TConfiguration(
-      TConfiguration.DEFAULT_MAX_FRAME_SIZE,
-      TConfiguration.DEFAULT_MAX_MESSAGE_SIZE,
-      Integer.MAX_VALUE);
+      TConfiguration.DEFAULT_MAX_FRAME_SIZE, TConfiguration.DEFAULT_MAX_MESSAGE_SIZE, Integer.MAX_VALUE);
 
-  private StubTTransport() {
-  }
+  private StubTTransport() {}
 
   @Override
-  public void checkReadBytesAvailable(long l) {
-  }
+  public void checkReadBytesAvailable(long l) {}
 
   @Override
   public boolean isOpen() {
@@ -56,8 +52,7 @@ public final class StubTTransport extends TTransport {
   }
 
   @Override
-  public void close() {
-  }
+  public void close() {}
 
   @Override
   public int read(byte[] bytes, int i, int i1) throws TTransportException {
@@ -75,6 +70,5 @@ public final class StubTTransport extends TTransport {
   }
 
   @Override
-  public void updateKnownMessageSize(long l) {
-  }
+  public void updateKnownMessageSize(long l) {}
 }

--- a/parquet-thrift/src/main/java/org/apache/parquet/thrift/StubTTransport.java
+++ b/parquet-thrift/src/main/java/org/apache/parquet/thrift/StubTTransport.java
@@ -1,0 +1,80 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.parquet.thrift;
+
+import org.apache.thrift.TConfiguration;
+import org.apache.thrift.transport.TTransport;
+import org.apache.thrift.transport.TTransportException;
+
+/**
+ * A stub transport which implements the minimum amount needed for range/depth
+ * validation within the thrift library to succeed.
+ */
+public final class StubTTransport extends TTransport {
+
+  public static final StubTTransport INSTANCE = new StubTTransport();
+
+  /**
+   * There's no limits on recursion depth.
+   */
+  private final TConfiguration conf = new TConfiguration(
+      TConfiguration.DEFAULT_MAX_FRAME_SIZE,
+      TConfiguration.DEFAULT_MAX_MESSAGE_SIZE,
+      Integer.MAX_VALUE);
+
+  private StubTTransport() {
+  }
+
+  @Override
+  public void checkReadBytesAvailable(long l) {
+  }
+
+  @Override
+  public boolean isOpen() {
+    return true;
+  }
+
+  @Override
+  public void open() throws TTransportException {
+    throw new TTransportException("unsupported");
+  }
+
+  @Override
+  public void close() {
+  }
+
+  @Override
+  public int read(byte[] bytes, int i, int i1) throws TTransportException {
+    throw new TTransportException("unsupported");
+  }
+
+  @Override
+  public void write(byte[] bytes, int i, int i1) throws TTransportException {
+    throw new TTransportException("unsupported");
+  }
+
+  @Override
+  public TConfiguration getConfiguration() {
+    return conf;
+  }
+
+  @Override
+  public void updateKnownMessageSize(long l) {
+  }
+}

